modules/imgproc/src/color_yuv.dispatch.cpp

```diff
@@ -406,12 +420,10 @@ void cvtColorTwoPlaneYUV2BGRpair( InputArray _ysrc, InputArray _uvsrc, OutputArr
 
     Mat ysrc = _ysrc.getMat(), uvsrc = _uvsrc.getMat();
 
-    CV_CheckEQ(ysrc.step, uvsrc.step, "");
```

Comment:
It makes sense to convert this into condition which selects corresponding HAL interface (with the same step value, or with different values)

Reply:
I think there is no point to support special cases, when we can fall fast to the general case.

Reply:
Point here is providing smooth migration and keep support for "old" HALs which don't have support for new API yet.

Reply:
This PR updates implementations of HAL wrappers for each available platform (AFAIK). Migration in this sense is not necessary.

Reply:
... stored with OpenCV source code.

For any other **external** HAL implementation (if any) this patch introduces regression.

Reply:
Legitimate :)

Reply:
Fixed.